stacktrace.go

```diff
@@ -20,23 +20,72 @@
 
 package zap
 
-import "runtime"
+import (
+	"runtime"
+	"strings"
+	"sync"
+
+	"go.uber.org/zap/internal/bufferpool"
+)
+
+var (
+	defaultUintptrSliceSize = 128
```

Comment:
Not a big issue since we're dealing with 4 byte values, but this is a huge call stack size. Maybe something like `32` would be more reasonable?